File: railway-oriented-validation/src/test/java/app/imperative/ImperativeValidationTest.java
package app.imperative;

import static ga.overfullstack.app.common.DataSet.EGG_CARTON;
import static ga.overfullstack.app.common.DataSet.EXPECTED_IMPERATIVE_VALIDATION_RESULTS;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

/**
 *
 *
 * <pre>
 * Problems:
 * - Complexity
 * - Mutation
 * . Non-sharable
 * - Unit-Testability
 * - Validation Jenga
 * </pre>
 */
class ImperativeValidationTest {
  @Test
  void cyclomaticCode() {
    final var badEggFailureBucketMap =
        ga.overfullstack.app.imperative.ImperativeValidation.validateEggCartonImperatively(
            EGG_CARTON.toJavaList());
    Assertions.assertEquals(EXPECTED_IMPERATIVE_VALIDATION_RESULTS, badEggFailureBucketMap);
  }
}

File: railway-oriented-validation-kt/src/test/kotlin/ga/overfullstack/app/imperative/ImperativeValidationTest.kt
package app.imperative

import app.common.eggCarton
import app.common.expectedImperativeValidationResults
import io.github.oshai.kotlinlogging.KotlinLogging
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

/* gakshintala created on 4/12/20 */
class ImperativeValidationTest {
  @Test
  fun cyclomaticCode() {
    val badEggFailureBucketMap =
      validateEggCartonImperatively(
        eggCarton.toMutableList()
      ) // sending a copy, a the iterator is common
    logger.info { badEggFailureBucketMap }
    Assertions.assertEquals(expectedImperativeValidationResults, badEggFailureBucketMap)
  }

  private val logger = KotlinLogging.logger {}
}
